coalib/collecting/Importers.py

```diff
@@ -7,28 +9,35 @@
 from coala_utils.ContextManagers import suppress_stdout
 from coala_utils.decorators import arguments_to_lists, yield_once
 
+# in Python > 3.4 this can be imported from importlib.utils
+from coalib.misc.Compatibility import module_from_spec
+
 
 def _import_module(file_path):
+    file_path = os.path.realpath(file_path)
     if not os.path.exists(file_path):
-        raise ImportError
+        raise ImportError("can't import non-existing file {!r}"
+                          .format(file_path))
 
+    # little HACK to get correct filename case on Windows
+    file_path = glob(file_path[:-1] + '[' + file_path[-1] + ']')[0]
     module_name = os.path.splitext(os.path.basename(file_path))[0]
-    module_dir = os.path.dirname(file_path)
-
-    if module_dir not in sys.path:
-        sys.path.insert(0, module_dir)
-
-    # Ugly inconsistency: Python will insist on correctly cased module names
-    # independent of whether the OS is case-sensitive or not.
-    # We want all cases to match though.
-    if platform.system() == 'Windows':  # pragma: nocover
-        for cased_file_path in os.listdir(module_dir):
-            cased_module_name = os.path.splitext(cased_file_path)[0]
-            if cased_module_name.lower() == module_name.lower():
-                module_name = cased_module_name
-                break
-
-    return __import__(module_name)
+    # don't replace any already imported modules
+    if module_name in sys.modules:
+        return sys.modules[module_name]
+
+    spec = spec_from_file_location(module_name, file_path)
+    if not spec:
+        raise ImportError("{!r} doesn't seem to be a python module"
+                          .format(file_path))
+
+    # load module, and add it to sys.modules for making inspect.getfile work
+    # with module members
+    module = sys.modules[module_name] = module_from_spec(spec)
+    spec.loader.exec_module(module)
+    # and extend sys.path for making pickling work
+    sys.path.insert(0, os.path.dirname(os.path.realpath(file_path)))
```

Comment:
Why are you doing this here, instead of keeping the existing code at https://github.com/coala/coala/pull/4210/files#r131307918